src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -61,104 +61,133 @@ static ColliderCylinderInit sCylinderInit = {
+    /* 0x7 */ STH_ANIM_PANIC,       // after buying Oceanside Spider House, can be found at bottom of slide,
+    /* 0x8 */ STH_ANIM_START,       // set in init, not an actual index to the array
+} EnSthAnimationIndexes;
+
+static AnimationHeader* sAnimationInfo[] = {
+    &gEnSthSignalAnim, &gEnSthBendDownAnim,   &gEnSthTalkWithHandUpAnim, &gEnSthWaitAnim,
+    &gEnSthLookUpAnim, &gEnSthLookAroundAnim, &gEnSthPleadAnim,          &gEnSthPanicAnim,
 };
 
-u16 D_80B6D1E8[] = { 0x1144, 0x1145, 0x1146 };
+// three slightly different variants of "Only a little time left, oh goddess please save me"
+static u16 sSthOceanspiderhousePanicText[] = { 0x1144, 0x1145, 0x1146 };
 
-u16 D_80B6D1F0[] = { 0x1139, 0x113E, 0x1143 };
+static u16 sSthFirstOceansideSpiderHouseGreetDialogue2[] = {
+    0x1139, // I thought I heard noises... I'm glad I found it early
+    0x113E, // I had no idea this place was here, perhaps it can keep me safe
+    0x1143  // There's no time we have to hide!
+};
 
-u16 D_80B6D1F8[] = { 0x1132, 0x113A, 0x113F };
+static u16 sSthFirstOceansideSpiderHouseGreetDialogue1[] = {
+    0x1132, // I thought I heard some noise.. is this some sort of underground shelter?
+    0x113A, // I had no idea there was a basement here..
+    0x113F  // I heard loud noises.. I never thought I'd find a place like this..
+};
 
-Vec3f D_80B6D200 = { 700.0f, 400.0f, 0.0f };
+static Vec3f sFocusOffset = { 700.0f, 400.0f, 0.0f };
 
-Color_RGB8 D_80B6D20C[] = {
-    { 190, 110, 0 }, { 0, 180, 110 }, { 0, 255, 0x50 }, { 255, 160, 60 }, { 190, 230, 250 }, { 240, 230, 120 },
+// Sth is hardcoded to only use element [1].
+static Color_RGB8 sShirtColors[] = {
+    { 190, 110, 0 },   // darker orange
+    { 0, 180, 110 },   // green
+    { 0, 255, 80 },    // lime green
+    { 255, 160, 60 },  // brighter orange
+    { 190, 230, 250 }, // white
+    { 240, 230, 120 }, // pale yellow
 };
 
 void EnSth_Init(Actor* thisx, PlayState* play) {
     s32 pad;
     EnSth* this = THIS;
     s32 objectId;
 
-    if (ENSTH_GET_100(&this->actor)) {
+    // this actor can draw two separate bodies that use different objects
+    if (STH_GET_SWAMP_BODY(&this->actor)) {
         objectId = Object_GetIndex(&play->objectCtx, OBJECT_AHG);
     } else {
         objectId = Object_GetIndex(&play->objectCtx, OBJECT_STH);
     }
-    this->unk_29E = objectId;
-    this->unk_29F = Object_GetIndex(&play->objectCtx, OBJECT_MASK_TRUTH);
+    this->mainObjectId = objectId;
+    this->maskOfTruthObjectId = Object_GetIndex(&play->objectCtx, OBJECT_MASK_TRUTH);
 
     Actor_SetScale(&this->actor, 0.01f);
     Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
     this->actor.colChkInfo.mass = MASS_IMMOVABLE;
     ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
 
-    this->unk_29C = 0;
-    this->unk_29A = 8;
+    this->sthFlags = STH_FLAG_CLEAR;
+    this->curAnimIndex = STH_ANIM_START;
```

Comment:
```suggestion
    this->animIndex = STH_ANIM_START;
```
Just `animIndex` is standard for this